test/models/patient_notification_eligibilty_test.rb

```diff
@@ -0,0 +1,400 @@
+# frozen_string_literal: true
+
+require 'test_case'
+
+class PatientNotificationEligibilityTest < ActiveSupport::TestCase
+  def setup
+    @default_purgeable_after = ADMIN_OPTIONS['purgeable_after']
+    @default_weekly_purge_warning_date = ADMIN_OPTIONS['weekly_purge_warning_date']
+    @default_weekly_purge_date = ADMIN_OPTIONS['weekly_purge_date']
+  end
+
+  def teardown
+    ADMIN_OPTIONS['purgeable_after'] = @default_purgeable_after
+    ADMIN_OPTIONS['weekly_purge_warning_date'] = @default_weekly_purge_warning_date
+    ADMIN_OPTIONS['weekly_purge_date'] = @default_weekly_purge_date
+  end
+
+  def assert_eligibility(patient, exp_eligibility)
+    assert_eligible(patient) if exp_eligibility
+    assert_ineligible(patient) unless exp_eligibility
+  end
+
+  def assert_eligible(patient)
+    eligible = !Patient.reminder_eligible.find_by(id: patient.id).nil?
+    puts "\nFailing eligible test with: #{format_patient_str(patient)}" unless eligible
```

Comment:
From my testing forcing tests to fail, I have not seen these `puts` statements. If you want this to be printed Assert can take an argument to output text on failure.